GVFS/GVFS.Service/Handlers/GetActiveRepoListHandler.cs

```diff
@@ -69,12 +69,18 @@ public void Run()
         private bool IsValidRepo(string repoRoot)
         {
             string gitBinPath = GVFSPlatform.Instance.GitInstallation.GetInstalledGitBinPath();
-            string hooksPath = ProcessHelper.WhereDirectory(GVFSPlatform.Instance.Constants.GVFSHooksExecutableName);
-            GVFSEnlistment enlistment = null;
+            string hooksPath = null;
 
             try
             {
-                enlistment = GVFSEnlistment.CreateFromDirectory(repoRoot, gitBinPath, hooksPath, authentication: null);
+                string hooksVersion = null;
+                GVFSEnlistment enlistment = null;

```

Comment:
Can this code that doesn't throw `InvalidRepoException` be moved back outside the `try`